sale_partner_version/models/res_partner.py

```diff
@@ -0,0 +1,30 @@
+# Copyright 2021 Akretion
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import models
+
+
+class ResPartner(models.Model):
+    _inherit = "res.partner"
+
+    def _version_impacted_columns(self):
+        result = super()._version_impacted_columns()
+        result += [
+            ("sale_order", "partner_id"),
```

Comment:
I do not think that we should archive the partner_id, I prefers to keep it on the active contact